src/overpass_api/statements/area_query.cc

```diff
@@ -287,8 +287,20 @@ void Area_Query_Statement::get_ranges
 
 bool Area_Constraint::delivers_data(Resource_Manager& rman)
 {
+  int counter = 0;
+
   if (!area->areas_from_input())
-    return false;
+  {
+    Block_Backend< Uint31_Index, Area_Skeleton > area_locations_db
+        (rman.get_area_transaction()->data_index(area_settings().AREAS));
+    for (Block_Backend< Uint31_Index, Area_Skeleton >::Flat_Iterator
+        it(area_locations_db.flat_begin());
+        !(it == area_locations_db.flat_end()); ++it)
+    {
+      if (area->get_submitted_id() == it.object().id.val())
```

Comment:
you mean just changing it to `it.handle().object().id.val()`, right? For me that seems to works as well.

It probably should also be changed on [line 244](https://github.com/tyrasd/Overpass-API/blob/200debd2dd02d513f5c31a8d5a826d9922f62008/src/overpass_api/statements/area_query.cc#L244)…